gui: add support for string/bytearray search with Python 3

sriemer opened the pull request:
Python 3 intentionally doesn't convert from bytes to unicode. This is why the search for string/bytearray freezes in GameConqueror and causes a debug message. So explicitly convert strings returned by scanmem as bytes to unicode.

This has been tested with Python 2.7 and 3.4 and is suitable for 0.15 stable.

Reported-by: calculuswhiz

sriemer commented:
I've tested this with Chromium B.S.U. and searched for "C h r o m i u m" as well as "43 20 68 20" which finds the same value which gets visible in the main menu.

coolwanglu commented on gui/GameConqueror.py:
```diff
@@ -905,7 +907,7 @@ def update_scan_result(self):
             self.scanresult_liststore.clear()
             for line in lines:
-                line = str(line)
+                line = str(line.decode("utf-8"))
```
`decode` already returns a `str`, so the outside `str` is not longer useful.

coolwanglu replied:
and `utf-8` is the default, so it could be simplified as `line = line.decode()`

sriemer replied:
I agree with `.decode()` without "utf-8" but your other remark is definitely not the case. This is with Python 2 without the `str()`:
````
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/share/gameconqueror/GameConqueror.py", line 889, in scan_thread_func
    self.update_scan_result()
  File "/usr/share/gameconqueror/GameConqueror.py", line 912, in update_scan_result
    (a, o, rt, v, t) = list(map(str.strip, line.split(',')[:5]))
TypeError: descriptor 'strip' requires a 'str' object but received a 'unicode'
````

coolwanglu replied:
Oh, maybe the input is from gtk. I'm not sure. 
So please just leave `str()`

sriemer replied:
Maybe the auto-conversion to str() is only valid for Python 3. At the other location I had to add str() for Python 2.

coolwanglu commented on gui/GameConqueror.py:
```diff
@@ -681,8 +681,10 @@ def bytes2value(self, typename, bytes):
+            bytes = str(bytes.decode("utf-8"))
             return repr('%s'%(bytes,))[1:-1]
         elif typename == 'bytearray':
+            bytes = bytes.decode("utf-8")
```
`bytes` is now a built-in type in python3, maybe we should rename it?

sriemer replied:
Would make sense to avoid confusion.

sriemer commented on gui/GameConqueror.py:
```diff
@@ -681,8 +681,10 @@ def bytes2value(self, typename, bytes):
         if typename in TYPENAMES_G2STRUCT:
             return struct.unpack(TYPENAMES_G2STRUCT[typename], bytes)[0]
         elif typename == 'string':
+            bytes = str(bytes.decode())
```
@coolwanglu Would the name `databytes` be okay for you?
````
--- a/gui/GameConqueror.py
+++ b/gui/GameConqueror.py
@@ -675,19 +675,19 @@ class GameConqueror():
         return None
 
     # parse bytes dumped by scanmem into number, string, etc.
-    def bytes2value(self, typename, bytes):
-        if bytes is None:
+    def bytes2value(self, typename, databytes):
+        if databytes is None:
             return None
         if typename in TYPENAMES_G2STRUCT:
-            return struct.unpack(TYPENAMES_G2STRUCT[typename], bytes)[0]
+            return struct.unpack(TYPENAMES_G2STRUCT[typename], databytes)[0]
         elif typename == 'string':
-            bytes = str(bytes.decode())
-            return repr('%s'%(bytes,))[1:-1]
+            databytes = str(databytes.decode())
+            return repr('%s'%(databytes,))[1:-1]
         elif typename == 'bytearray':
-            bytes = bytes.decode()
-            return ' '.join(['%02x'%ord(i) for i in bytes])
+            databytes = databytes.decode()
+            return ' '.join(['%02x'%ord(i) for i in databytes])
         else:
-            return bytes
+            return databytes
     
     def scan_for_addr(self, addr):
         bits = self.get_pointer_width()
````

coolwanglu replied:
Looks good. Please use it.

sriemer commented:
Awesome, forcefully pushing works without closing the PR. I've moved the corrections of the decode()s to the respective commits instead of fixing them one or two commits afterwards. I've also rebased the branch to upstream/master so that we are up-to-date again and can continue this one.

sriemer commented:
@coolwanglu Okay, now I also have the rename. We should be ready to merge now.

sriemer commented:
I've tested this with both Python versions again, both string and bytearray, both initial and second search. It works.
